mesonbuild/interpreter.py

```diff
@@ -3616,8 +3662,32 @@ def dependency_fallback(self, name, display_name, kwargs):
             'required': required,
         }
         self.do_subproject(dirname, 'meson', sp_kwargs)
+        self.override_dependencies_from_wrap(dirname, kwargs)
```

Comment:
I think that's specific to usage as fallback because it uses for example "native" kwarg for the cache identifier that won't be set when calling subproject(). It's really when you return one dependency object that you want to ensure it picks the fallback too for any other dependency provided by that subproject.